dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layerparam/initializer/ConvolutionalLayerParameterInitializer.java

```diff
@@ -109,12 +118,21 @@ public int getIndex() {
    * @return shape of output
    */
   private int[] computeOutputShape() {
-    final int[] computedShape = new int[2];
-    if (inputShape.length != 2) {
+    final int[] computedShape = new int[3];
+    if (inputShape.length != 2 && inputShape.length != 3) {
       throw new IllegalArgumentException("Unsupported input dimensions: " + inputShape.length);
     }
-    computedShape[0] = (int) Math.ceil((float) (inputShape[0] - kernelHeight + 2 * paddingHeight) / strideHeight) + 1;
-    computedShape[1] = (int) Math.ceil((float) (inputShape[1] - kernelWidth + 2 * paddingWidth) / strideWidth) + 1;
+    if (inputShape.length == 2) {
+      computedShape[0] = (int) Math.ceil((float) (inputShape[0] - kernelHeight + 2 * paddingHeight) / strideHeight) + 1;
+      computedShape[1] = (int) Math.ceil((float) (inputShape[1] - kernelWidth + 2 * paddingWidth) / strideWidth) + 1;
+    } else {
+      computedShape[0] = (int) Math.ceil((float) (inputShape[1] - kernelHeight + 2 * paddingHeight) / strideHeight) + 1;
+      computedShape[1] = (int) Math.ceil((float) (inputShape[2] - kernelWidth + 2 * paddingWidth) / strideWidth) + 1;
+    }
+    computedShape[2] = numOutput;
```

Comment:
Why don't we use `computeShape[0]` as the number of output channels for the consistency? For the input shape, `inputShape[0]`  indicates the number of input channels.